core/controllers/classifier_test.py

```diff
@@ -168,10 +167,10 @@ def mock_get_classifier_training_job_by_id(_):
 
                 # Check that there are no sent emails to either
                 # email address before posting json.
-                messages = self.mail_stub.get_sent_messages(
+                messages = self.email_services_mock.mock_get_sent_messages(
```

Comment:
This should call the regular methods. (email_services.get_sent_messages() etc.). EmailTestBase should take care of the swap at the platform layer so that calling the regular method does the right thing.

If you call a mock, then we end up with a "skew" problem again, where the code you are testing is not the same as the code that runs in production.

This needs to be fixed in many places elsewhere, too.

Reply:
Ah, sorry, you're right here, my bad! Scrap the previous comment, it doesn't apply for this particular method (though see the other notes for perhaps not having this as a separate mock class).